Source/FluentSpreadsheets.GoogleSheets/Factories/RenderCommandFactory.cs

```diff
@@ -0,0 +1,72 @@
+﻿using FluentSpreadsheets.GoogleSheets.Exceptions;
+using FluentSpreadsheets.GoogleSheets.Rendering;
+using Google.Apis.Sheets.v4;
+using Google.Apis.Sheets.v4.Data;
+
+namespace FluentSpreadsheets.GoogleSheets.Factories;
+
+public class RenderCommandFactory : IRenderCommandFactory
+{
+    private readonly SheetsService _sheetsService;
+
+    public RenderCommandFactory(SheetsService sheetsService)
+    {
+        _sheetsService = sheetsService;
+    }
+
+    public async Task<GoogleSheetRenderCommand> CreateAsync(
+        string spreadsheetId,
+        string title,
+        IComponent component,
+        CancellationToken cancellationToken = default)
+    {
+        int id = await GetSheetId(spreadsheetId, title, cancellationToken);
+        return new GoogleSheetRenderCommand(spreadsheetId, id, title, component);
+    }
+
+    public async Task<GoogleSheetRenderCommand> CreateAsync(
+        string spreadsheetId,
+        int id,
+        IComponent component,
+        CancellationToken cancellationToken = default)
+    {
+        string title = await GetSheetTitle(spreadsheetId, id, cancellationToken);
+        return new GoogleSheetRenderCommand(spreadsheetId, id, title, component);
+    }
+
+    private async Task<int> GetSheetId(string spreadsheetId, string title, CancellationToken cancellationToken)
+    {
+        IList<Sheet> sheets = await GetSheetsAsync(spreadsheetId, cancellationToken);
+
+        Sheet googleSheet = sheets.FirstOrDefault(s => s.Properties.Title == title)
+                            ?? throw new GoogleSheetException($"Sheet with title {title} does not exist");
+
+        int? sheetId = googleSheet.Properties.SheetId;
+
+        if (sheetId is null)
+        {
+            throw new GoogleSheetException("Sheet id does not exist");
+        }
+
+        return sheetId.Value;
+    }
+
+    private async Task<string> GetSheetTitle(string spreadsheetId, int id, CancellationToken cancellationToken)
+    {
+        IList<Sheet> sheets = await GetSheetsAsync(spreadsheetId, cancellationToken);
+
+        Sheet googleSheet = sheets.FirstOrDefault(s => s.Properties.SheetId == id)
+                            ?? throw new GoogleSheetException($"Sheet with id {id} does not exist");
```

Comment:
Don't really think that null coalescing is the best option for null check here (same with the method above), it is pretty bulky, so I'd rather user just `if (googleSheet is null)...`